Assets/Scripts/EOSLobbyManager.cs

```diff
@@ -1529,65 +1535,137 @@ public void PressToTalk(KeyCode PTTKeyCode, OnLobbyCallback TogglePressToTalkCom
+        /// If CurrentLobby has populated its <see cref="Lobby.Members"/> list,
+        /// the local user's current mute setting is reversed, and then applied.
+        /// If the Members list isn't available, this action will attempt to mute the target.
+        /// If the local user has another user in the Lobby muted, the local user won't hear the muted user, but other members may still be able to.
+        /// If the local user mutes themself, other users won't be able to hear the local user.
         /// </summary>
         /// <param name="targetUserId">Target <c>ProductUserId</c> to mute or unmute</param>
         /// <param name="ToggleMuteCompleted">Callback when toggle mute is completed</param>
         public void ToggleMute(ProductUserId targetUserId, OnLobbyCallback ToggleMuteCompleted)
         {
+            // If the user can be found in CurrentLobby.Members, set this value to the opposite of the user's current mute value
+            // Otherwise, always attempt to mute
+            bool shouldAudioBeEnabled = false;
+
             RTCInterface rtcHandle = EOSManager.Instance.GetEOSRTCInterface();
             RTCAudioInterface rtcAudioHandle = rtcHandle.GetAudioInterface();
 
-            foreach(LobbyMember lobbyMember in CurrentLobby.Members)
+            LobbyMember lobbyMember = GetCurrentLobby().Members.Find(x => x.ProductId == targetUserId);
+            if (lobbyMember != null)
             {
-                // Find the correct lobby member
-                if(lobbyMember.ProductId != targetUserId)
-                {
-                    continue;
-                }
+                // If the target is muted, "IsLocalMuted" is true,
+                // since this is a toggle, we want to take the opposite of that
+                // therefore this would become "true", which will result in the target becoming unmuted
+                shouldAudioBeEnabled = lobbyMember.RTCState.IsLocalMuted;
+            }
+
+            MuteMember(shouldAudioBeEnabled, targetUserId, ToggleMuteCompleted);
+        }
+
+        /// <summary>
+        /// Set the local mute status of a member in the lobby.
+        /// If the local user has another user in the Lobby muted, the local user won't hear the muted user, but other members may still be able to.
+        /// If the local user mutes themself, other users won't be able to hear the local user.
+        /// </summary>
+        /// <param name="shouldAudioBeEnabled">
+        /// If true, audio "becomes enabled", unmuting the target.
+        /// If false, audio "becomes disabled", muting the target.
+        /// </param>
+        /// </param>
+        /// <param name="targetProductUserId">The Product User ID of the target member to mute</param>
+        /// <param name="MuteMemberCompleted">Callback when MuteMember is completed</param>
+        public void MuteMember(bool shouldAudioBeEnabled, ProductUserId targetProductUserId, OnLobbyCallback MuteMemberCompleted)
```

Comment:
I think contextually, it makes sense to rename this function signature to `SetIsMemberMuted`, or something that is not just `Mute`, since this function can actually _unmute_ a member, so the name is misleading as-is.

Reply:
Imma try `SetMemberMuteStatus`.